src/PointwiseFunctions/MathFunctions/Sum.cpp

```diff
@@ -0,0 +1,284 @@
+}
+
+template <typename Fr>
+DataVector Sum<1, Fr>::operator()(const DataVector& x) const noexcept {
+  return apply_call_operator(x);
+}
+
+template <typename Fr>
+double Sum<1, Fr>::first_deriv(const double& x) const noexcept {
+  return apply_first_deriv(x);
+}
+
+template <typename Fr>
+DataVector Sum<1, Fr>::first_deriv(const DataVector& x) const noexcept {
+  return apply_first_deriv(x);
+}
+
+template <typename Fr>
+double Sum<1, Fr>::second_deriv(const double& x) const noexcept {
+  return apply_second_deriv(x);
+}
+
+template <typename Fr>
+DataVector Sum<1, Fr>::second_deriv(const DataVector& x) const noexcept {
+  return apply_second_deriv(x);
+}
+
+template <typename Fr>
+double Sum<1, Fr>::third_deriv(const double& x) const noexcept {
+  return apply_third_deriv(x);
+}
+
+template <typename Fr>
+DataVector Sum<1, Fr>::third_deriv(const DataVector& x) const noexcept {
+  return apply_third_deriv(x);
+}
+
+template <typename Fr>
+template <typename T>
+T Sum<1, Fr>::apply_call_operator(const T& x) const noexcept {
+  return math_function_a_->operator()(x) + math_function_b_->operator()(x);
+}
+
+template <typename Fr>
+template <typename T>
+T Sum<1, Fr>::apply_first_deriv(const T& x) const noexcept {
+  return math_function_a_->first_deriv(x) + math_function_b_->first_deriv(x);
+}
+
+template <typename Fr>
+template <typename T>
+T Sum<1, Fr>::apply_second_deriv(const T& x) const noexcept {
+  return math_function_a_->second_deriv(x) + math_function_b_->second_deriv(x);
+}
+
+template <typename Fr>
+template <typename T>
+T Sum<1, Fr>::apply_third_deriv(const T& x) const noexcept {
+  return math_function_a_->third_deriv(x) + math_function_b_->third_deriv(x);
+}
+
+template <typename Fr>
+void Sum<1, Fr>::pup(PUP::er& p) {
+  MathFunction<1, Fr>::pup(p);
+  p | math_function_a_;
+  p | math_function_b_;
+}
+
+template <size_t VolumeDim, typename Fr>
+Sum<VolumeDim, Fr>::Sum(
+    std::unique_ptr<MathFunction<VolumeDim, Fr>> math_function_a,
+    std::unique_ptr<MathFunction<VolumeDim, Fr>> math_function_b) noexcept
+    : math_function_a_(std::move(math_function_a)),
+      math_function_b_(std::move(math_function_b)) {}
+
+template <size_t VolumeDim, typename Fr>
+template <typename T>
+Scalar<T> Sum<VolumeDim, Fr>::apply_call_operator(
+    const tnsr::I<T, VolumeDim, Fr>& x) const noexcept {
+  Scalar<T> result = math_function_a_->operator()(x);
+  get(result) += get(math_function_b_->operator()(x));
+  return result;
+}
+
+template <size_t VolumeDim, typename Fr>
+template <typename T>
+tnsr::i<T, VolumeDim, Fr> Sum<VolumeDim, Fr>::apply_first_deriv(
+    const tnsr::I<T, VolumeDim, Fr>& x) const noexcept {
+  const tnsr::i<T, VolumeDim, Fr>& first_deriv_a =
+      math_function_a_->first_deriv(x);
+  const tnsr::i<T, VolumeDim, Fr>& first_deriv_b =
+      math_function_b_->first_deriv(x);
+  return TensorExpressions::evaluate<ti_a_t>(first_deriv_a(ti_a) +
+                                             first_deriv_b(ti_a));
```

Comment:
for (size_t tensor_index = 0; tensor_index < x.size(); ++tensor_index) {
   first_deriv_a[i] += ....
}